pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -550,3 +557,52 @@ func (r *ReconcileAWSFederatedAccountAccess) cleanFederatedRoles(reqLogger logr.
 
 	return nil
 }
+
+func (r *ReconcileAWSFederatedAccountAccess)  deleteNonAttachedCustomPolicy(reqLogger logr.Logger, awsClient awsclient.Client, federatedRoleCR *awsv1alpha1.AWSFederatedRole) error {
+
+	var policyMarker *string
+	// Paginate through custom policies
+	for {
+		policyListOutput, err := awsClient.ListPolicies(&iam.ListPoliciesInput{Scope: aws.String("Local"), Marker: policyMarker})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
```

Comment:
Since we are not doing any filtering here we shouldn't be using the switch we should just return the error. If we need more granular error handling in the future we can add it at that point.

Reply:
This makes it easily extensible in the future. I.e. line 472 above.